internal/guest/runtime/hcsv2/nvidia_utils.go

```diff
@@ -8,31 +8,35 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/pkg/errors"
 
 	"github.com/Microsoft/hcsshim/cmd/gcstools/generichook"
+	specInternal "github.com/Microsoft/hcsshim/internal/guest/spec"
 	"github.com/Microsoft/hcsshim/internal/guest/storage/pci"
 	"github.com/Microsoft/hcsshim/internal/hooks"
 	"github.com/Microsoft/hcsshim/pkg/annotations"
 )
 
-const nvidiaDebugFilePath = "/nvidia-container.log"
+const nvidiaHookDebugFile = "nvidia-hook.log"
+const nvidiaDebugFilePath = "nvidia-container.log"
 
 const nvidiaToolBinary = "nvidia-container-cli"
 
 // described here: https://github.com/opencontainers/runtime-spec/blob/39c287c415bf86fb5b7506528d471db5405f8ca8/config.md#posix-platform-hooks
 // addNvidiaDeviceHook builds the arguments for nvidia-container-cli and creates the prestart hook
-func addNvidiaDeviceHook(ctx context.Context, spec *oci.Spec) error {
+func addNvidiaDeviceHook(ctx context.Context, spec *oci.Spec, sbid string) error {
 	genericHookBinary := "generichook"
 	genericHookPath, err := exec.LookPath(genericHookBinary)
 	if err != nil {
 		return errors.Wrapf(err, "failed to find %s for container device support", genericHookBinary)
 	}
 
-	debugOption := fmt.Sprintf("--debug=%s", nvidiaDebugFilePath)
+	toolDebugPath := filepath.Join(specInternal.SandboxRootDir(sbid), nvidiaDebugFilePath)
```

Comment:
quick question: is it ok that we'll have only one tool/cli log file even if multiple containers may have this hook? (I see that it was like that before as well)

Reply:
Good point, I tested it out and it would definitely be clearer to have the logs separated by container. I'll add that and update here.

Reply:
Updated.